packages/next/src/server/base-server.ts

```diff
@@ -3027,39 +3027,47 @@ export default abstract class Server<
     }
 
     // If we're not in minimal mode and the cache entry that was returned was a
-    // app page fallback, then we need to kick off the dynamic shell generation.
+    // app page fallback, then we need to kick off the route shell generation.
     if (
       ssgCacheKey &&
       !this.minimalMode &&
       isRoutePPREnabled &&
       this.nextConfig.experimental.pprFallbacks &&
       cacheEntry.value?.kind === CachedRouteKind.APP_PAGE &&
       cacheEntry.isFallback &&
-      !isOnDemandRevalidate
+      !isOnDemandRevalidate &&
+      process.env.NEXT_RUNTIME !== 'edge'
```

Comment:
why not edge?

Reply:
Edge doesn't support ISR! So it's needed to perform the check here.